tour_model_eval/trips_in_bins_and_clusters.ipynb

```diff
@@ -0,0 +1,344 @@
+    "import logging\n",
+    "import folium\n",
+    "import math"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "caroline-figure",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import emission.storage.timeseries.abstract_timeseries as esta"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "combined-magazine",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# copied from mode_purpose_share.ipynb\n",
+    "# We select participants from real data to be users at this time\n",
+    "participant_uuid_obj = list(edb.get_profile_db().find({\"install_group\": \"participant\"}, {\"user_id\": 1, \"_id\": 0}))\n",
+    "all_users = [u[\"user_id\"] for u in participant_uuid_obj]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "adaptive-supplement",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "logger = logging.getLogger()\n",
+    "logger.setLevel(logging.DEBUG)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "hollow-prefix",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "logging.debug('There are %s users, we will plot the graphs for one of them' % len(all_users))\n",
+    "\n",
+    "user = all_users[1]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "prospective-conducting",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import emission.analysis.modelling.tour_model.cluster_pipeline as eamtc"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "finnish-istanbul",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# we should experiment with different values here\n",
+    "radius = 300"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "frequent-slovakia",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import emission.analysis.modelling.tour_model.similarity as similarity"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bronze-switch",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import emission.analysis.modelling.tour_model.cluster_pipeline as pipeline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "consecutive-interface",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import emission.analysis.modelling.tour_model.featurization as featurization"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dependent-machinery",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "trips = pipeline.read_data(uuid=user)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "round-death",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "bin_trips, bins = pipeline.remove_noise(trips, radius)"
```

Comment:
We should revisit the level at which we are calling the cluster pipeline functions for visualization.
For example, `remove_noise` doesn't only bin the values, it also removes all the uncommon bins.
So we can't visualize/work with those bins because we never see them.

That is fine if we agree with the novelty detection method; not so great if we don't.
Similar considerations apply to calling `cluster` versus the featurization code directly